### PR TITLE
Enable `revive` linter

### PR DESCRIPTION
**Pull request by seeforschauer**

### CODE REVIEW

**Review by w84thesun — APPROVED** [review]
LGTM.

File: .golangci.yml [hunk 1]
```diff
@@ -50,8 +50,8 @@ linters-settings:
   goimports:
     local-prefixes: github.com/FerretDB/FerretDB
   gosimple:
-    go: '1.18'
-    checks: ['all']
+    go: "1.18"
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply .golangci.yml:53]: I believe so!

**Review by rumyantseva — DISMISSED** [review]
LGTM! Thanks!

File: .golangci.yml [hunk 2]
```diff
@@ -96,6 +96,7 @@ linters:
     - staticcheck
     - unused
     - whitespace
+    - revive
```
Comment (AlekSi) [inline .golangci.yml:99]: Please keep the list sorted.

Also, please add revive's configuration above

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: .golangci.yml (hunk 2 above)
Comment (seeforschauer) [reply .golangci.yml]: > Oh, and it should be added to one block and removed from another block in `.golangci-new.yml` too

added.

> Also, please add revive's configuration above

added.

Comment (AlekSi) [reply .golangci.yml]: It was removed from one block, but wasn't added into another

File: .golangci.yml [hunk 3]
```diff
@@ -54,6 +54,11 @@ linters-settings:
     checks: ["all"]
   # govet
   # ineffassign
+  revive:
+    ignore-generated-header: true
+    severity: warning
+    enableAllRules: true
+
```
Comment (AlekSi) [inline .golangci.yml:61]: There are no empty lines between other configurations

File: .golangci-new.yml [hunk 4]
```diff
@@ -100,13 +99,13 @@ linters:
...
-    - nlreturn    # too annoying
-    - scopelint   # deprecated
+    - golint # deprecated
+    - interfacer # deprecated
+    - maligned # deprecated
+    - nlreturn # too annoying
+    - scopelint # deprecated
     - testpackage # we test internals
-    - wsl         # too annoying
+    - wsl # too annoying
```
Comment (AlekSi) [inline .golangci-new.yml:108]: Please revert those changes

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: .golangci-new.yml [hunk 5]
```diff
@@ -77,7 +77,6 @@ linters:
     - prealloc
     - predeclared
     - promlinter
-    - revive
```
[the remark this answers is not in the document]
Comment (seeforschauer) [reply .golangci-new.yml:80]: > It should be added in the block above

now I see, from one disabled to another disabled.

File: .golangci.yml (hunk 3 above)
Comment (seeforschauer) [reply .golangci.yml]: removed.

File: .golangci.yml [hunk 6]
```diff
@@ -90,6 +95,7 @@ linters:
     - gosimple
     - govet
     - ineffassign
+    - revive
```
[the remark this answers is not in the document]
Comment (seeforschauer) [reply .golangci.yml]: moved before s.

File: .golangci-new.yml (hunk 4 above)
Comment (seeforschauer) [reply .golangci-new.yml]: done.

File: .golangci.yml [hunk 7]
```diff
@@ -54,6 +54,11 @@ linters-settings:
     checks: ["all"]
   # govet
   # ineffassign
+  revive:
```
[the remark this answers is not in the document]
Comment (AlekSi) [reply .golangci.yml:57]: That wasn't done

File: .golangci.yml (hunk 2 above)
Comment (seeforschauer) [reply .golangci.yml]: > It was removed from one block, but wasn't added into another

yes, it was not clear that we manage different lists of disabled linters.

File: .golangci-new.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (AlekSi) [reply .golangci-new.yml:80]: Please keep that list above sorted alphabetically :)

File: .golangci.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (seeforschauer) [reply .golangci.yml:57]: sorry found n another place.

[the remark this answers is not in the document]
Comment (seeforschauer) [reply .golangci.yml]: updated.

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**